### ISSUE TITLE
Building - features are added to "default" Unity layer - I need them to be in custom layer though

### ISSUE THREAD

**Issue by gisinator** [mapbox/mapbox-unity-sdk#1804]
_Note: If this is a bug or support ticket, please provide the following information:_
* Unity version: 2020.3.16f1
* Mapbox SDK version: 2.1.1 
* The platform you're building to: Win 10
* A description of what you're trying to do: In my scene there is two cameras. One camera rendering random stuff plus a second camera rendering all mapbox-related stuff. I can easily set this "mapbox" layer to all my mapbox-gameobjects and also tick "add to Unity layer" in "terrain" settings of abstract-map. But when features are added in runtime, those features are in "default" layer so my mapbox-camera does not render them.
* Steps to recreate the bug if appropriate:
* 1) create a custom unity layer
* 2) set all mapbox-gameobjects to this custom layer
* 3) tick "add to unity layer" in "terrain" section of abstractMap.cs
* 4) add "building" feature
* 5) play - features are NOT rendered in custom layer but in default layer

Any idea how I can make features are added on my custom layer? Thanks :)

**Comment by brnkhy** [comment]
@gisinator you are using layer settings for terrain object. If you check the tile/terrain gameobject, it should be on the layer you set.
to change generated objects' layer, you should use `LayerModifier` gameobject. if you add it to `gameobject modifiers` list, it'll set layer of every feature/object using that style.

**Comment by ashokdube** [comment]
Ok

**Comment by gisinator** [comment]
> @gisinator you are using layer settings for terrain object. If you check the tile/terrain gameobject, it should be on the layer you set. to change generated objects' layer, you should use `LayerModifier` gameobject. if you add it to `gameobject modifiers` list, it'll set layer of every feature/object using that style.

Thank you for your reply @brnkhy . I am not quite sure whether I understood correctly, here is what I just did:

1) In AbstractMap.cs, under "TERRAIN" section, I tick "Add to Unity layer" and select my "Mapbox" layer.
2) As a result, the tile/terrain gameobject is in fact on this layer. Great.
3) In "FEATURES" section, I add "Buildings" map feature. 
4) Following your reply, under "Behavior Modifiers" --> "Game Object Modifiers", I click "Add New" and choose "NewLayerModifier". I dont see any settings here, so that's all I do.
5) Unfortunately, generated buildings are NOT in my "Mapbox" layer, even though they are children of my tile/terrain objects that ARE in "Mapbox" layer. Instead. those buildings are in "Default" layer

What am I missing here? Did I misunderstand you?

Appreciate any further help :)

**edit:** Found the solution. Instead of "Add New" modifier, I had to click on "Add existing", where I could set my custom layer. Great. thank you, @brnkhy :)
